.github/workflows/CI.yml

```diff
@@ -67,14 +80,38 @@ jobs:
   integrationTests:
     runs-on: ubuntu-18.04
 
-    steps:
+    steps:         
     - name: Checkout code
       uses: actions/checkout@v1
+    - name: Cache ivy2
+      uses: actions/cache@v1
+      with:
+        path: /home/runner/.ivy2
+        key: cmwell-it-ivy2-${{ hashFiles('**/build.sbt') }}
+        restore-keys: cmwell-it-ivy2
+    - name: Cache coursier
+      uses: actions/cache@v1
+      with:
+        path: /home/runner/.cache/coursier
+        key: cmwell-it-cache-${{ hashFiles('**/build.sbt') }}
+        restore-keys: cmwell-it-cache
+    - name: Cache Scala 2.13
+      uses: actions/cache@v1
+      with:
+        path: /home/runner/work/CM-Well/CM-Well/scala-2.13.1
+        key: scala2_13_1
     - name: Install Scala
       run: |
-        wget https://downloads.lightbend.com/scala/2.13.1/scala-2.13.1.tgz
-        sudo tar -xf scala-2.13.1.tgz
-        sudo ln -s `pwd`/scala-2.13.1/bin/scala /usr/local/bin/scala
+        scalaVersion="2.13.1"
+        if [ ! -d "`pwd`/scala-$scalaVersion" ]; then
+          echo "`pwd`"
+          echo "Scala does not exist! Going to download"
+          wget https://downloads.lightbend.com/scala/$scalaVersion/scala-$scalaVersion.tgz
+          sudo tar -xf scala-$scalaVersion.tgz
```

Comment:
extract always to the same place.
Don't forget to delete (rm) the directory before (why directory is called like this and not "folder" :smile: )